test/ng/compileSpec.js

```diff
@@ -12111,6 +12142,890 @@ describe('$compile', function() {
+          $provide.value('$$sanitizeUri', $$sanitizeUri);
+        });
+        inject(function($compile, $rootScope) {
+          element = $compile('<img ng-prop-src="testUrl"></img>')($rootScope);
+          $rootScope.testUrl = 'someUrl';
+
+          $$sanitizeUri.and.returnValue('someSanitizedUrl');
+          $rootScope.$apply();
+          expect(element.prop('src')).toMatch(/^http:\/\/.*\/someSanitizedUrl$/);
+          expect($$sanitizeUri).toHaveBeenCalledWith($rootScope.testUrl, true);
+        });
+      });
+
+      it('should not use $$sanitizeUri with trusted values', function() {
+        var $$sanitizeUri = jasmine.createSpy('$$sanitizeUri').and.throwError('Should not have been called');
+        module(function($provide) {
+          $provide.value('$$sanitizeUri', $$sanitizeUri);
+        });
+        inject(function($compile, $rootScope, $sce) {
+          element = $compile('<img ng-prop-src="testUrl"></img>')($rootScope);
+          // Assigning javascript:foo to src makes at least IE9-11 complain, so use another
+          // protocol name.
+          $rootScope.testUrl = $sce.trustAsMediaUrl('untrusted:foo();');
+          $rootScope.$apply();
+          expect(element.prop('src')).toBe('untrusted:foo();');
+        });
+      });
+    });
+
+    ['img', 'source'].forEach(function(srcsetElement) {
+      describe(srcsetElement + '[srcset] sanitization', function() {
+        it('should not error if srcset is undefined', inject(function($compile, $rootScope) {
+          element = $compile('<' + srcsetElement + ' ng-prop-srcset="testUrl"></' + srcsetElement + '>')($rootScope);
+          // Set srcset to a value
+          $rootScope.testUrl = 'http://example.com/';
+          $rootScope.$digest();
+          expect(element.prop('srcset')).toBe('http://example.com/');
+
+          // Now set it to undefined
+          $rootScope.testUrl = '';
+          $rootScope.$digest();
+          expect(element.prop('srcset')).toBe('');
+        }));
+
+        it('should NOT require trusted values for whitelisted values', inject(function($rootScope, $compile, $sce) {
+          element = $compile('<' + srcsetElement + ' ng-prop-srcset="testUrl"></' + srcsetElement + '>')($rootScope);
+          $rootScope.testUrl = 'http://example.com/image.png'; // `http` is whitelisted
+          $rootScope.$digest();
+          expect(element.prop('srcset')).toEqual('http://example.com/image.png');
+        }));
+
+        it('should accept trusted values, if they are also whitelisted', inject(function($rootScope, $compile, $sce) {
+          element = $compile('<' + srcsetElement + ' ng-prop-srcset="testUrl"></' + srcsetElement + '>')($rootScope);
+          $rootScope.testUrl = $sce.trustAsUrl('http://example.com');
+          $rootScope.$digest();
+          expect(element.prop('srcset')).toEqual('http://example.com');
+        }));
+
+        it('does not work with trusted values', inject(function($rootScope, $compile, $sce) {
```

Comment:
_does_ --> _should_
_not_ --> _NOT_? 😛

Reply:
Again this was copied from the attr version. Shall I change both?